tests/glsl-intrinsic/shader-subgroup/shader-subgroup-shuffle.slang

```diff
@@ -0,0 +1,139 @@
+#ifdef TEST_when_subgroupShuffleXor_is_implemented
+        && subgroupShuffleXor(gvec(T(1)), 1) == gvec(T(1))
+#endif // #ifdef TEST_when_subgroupShuffleXor_is_implemented
+        ;
+}
+
+__generic<T : __BuiltinFloatingPointType>
+bool test1ShuffleX() {
+    return true
+        && subgroupShuffle(T(1), 1) == T(1)
+#if !defined(TARGET_CUDA) && !defined(TARGET_HLSL)
+        && subgroupShuffleXor(T(1), 1) == T(1)
+#endif // #if !defined(TARGET_CUDA) && !defined(TARGET_HLSL)
+        ;
+}
+__generic<T : __BuiltinFloatingPointType, let N : int>
+bool testVShuffleX() {
+    typealias gvec = vector<T, N>;
+
+    return true
+        && subgroupShuffle(gvec(T(1)), 1) == gvec(T(1))
+#if !defined(TARGET_CUDA) && !defined(TARGET_HLSL)
+        && subgroupShuffleXor(gvec(T(1)), 1) == gvec(T(1))
+#endif // #if !defined(TARGET_CUDA) && !defined(TARGET_HLSL)
+        ;
+}
+bool testShuffleX() {
+    return true
+        && test1ShuffleX<float>()
+        && testVShuffleX<float, 2>()
+        && testVShuffleX<float, 3>()
+        && testVShuffleX<float, 4>()
+        && test1ShuffleX<double>() // WARNING: intel GPU's lack FP64 support
+        && testVShuffleX<double, 2>()
+        && testVShuffleX<double, 3>()
+        && testVShuffleX<double, 4>()
+        && test1ShuffleX<half>() 
+        && testVShuffleX<half, 2>()
+        && testVShuffleX<half, 3>()
+        && testVShuffleX<half, 4>()
+        && test1ShuffleX<int>()
+        && testVShuffleX<int, 2>()
+        && testVShuffleX<int, 3>()
+        && testVShuffleX<int, 4>()
+        && test1ShuffleX<int8_t>() 
+        && testVShuffleX<int8_t, 2>()
+        && testVShuffleX<int8_t, 3>()
+        && testVShuffleX<int8_t, 4>()
+        && test1ShuffleX<int16_t>() 
+        && testVShuffleX<int16_t, 2>()
+        && testVShuffleX<int16_t, 3>()
+        && testVShuffleX<int16_t, 4>()
+        && test1ShuffleX<int64_t>() 
+        && testVShuffleX<int64_t, 2>()
+        && testVShuffleX<int64_t, 3>()
+        && testVShuffleX<int64_t, 4>()
+        && test1ShuffleX<uint>()
+        && testVShuffleX<uint, 2>()
+        && testVShuffleX<uint, 3>()
+        && testVShuffleX<uint, 4>()
+        && test1ShuffleX<uint8_t>() 
+        && testVShuffleX<uint8_t, 2>()
+        && testVShuffleX<uint8_t, 3>()
+        && testVShuffleX<uint8_t, 4>()
+        && test1ShuffleX<uint16_t>() 
+        && testVShuffleX<uint16_t, 2>()
+        && testVShuffleX<uint16_t, 3>()
+        && testVShuffleX<uint16_t, 4>()
+        && test1ShuffleX<uint64_t>() 
+        && testVShuffleX<uint64_t, 2>()
+        && testVShuffleX<uint64_t, 3>()
+        && testVShuffleX<uint64_t, 4>()
+        && test1ShuffleX<bool>()
+        && testVShuffleX<bool, 2>()
+        && testVShuffleX<bool, 3>()
+        && testVShuffleX<bool, 4>()
+        ;
+}
+
+
+void computeMain()
+{
+
+    outputBuffer.data[0] = true
+        && testShuffleX()
+        ;
+
+    // CHECK_GLSL: void main(
+    // CHECK_SPV: OpEntryPoint
```

Comment:
These checks are too simplistic. At least verify the expected SPIRV opcodes are existent.

Reply:
with generics the string checking is quite unreliable; for the larger testing code base, most code does not check for every op code (especially for bigger templated tests).

I can probably still add, but it will take a while to add to all of the files

Reply:
I agree that we should check the expected SPIRV opcodes.
We can probably do it as a new task later.

For now, the following test will cover the spir-v test.
```
//TEST(compute, vulkan):COMPARE_COMPUTE(filecheck-buffer=BUF):-vk -compute -entry computeMain -allow-glsl -emit-spirv-directly
```